### PR TITLE
fixes most bugs

### PR DESCRIPTION
**Pull request by charles-cowart**
This PR fixes most errors in testing that were occurring before and after integrating Justin's changes. Three errors remain:
_test_abundance_read_count()
test_read_count_to_cell_count()
test_same_as_R()_

All with the same error message:

_raise KeyError(f"None of [{key}] are in the [{axis_name}]")
KeyError: "None of [Index(['G000005825', 'G000006605', 'G000006725', 'G000006745', 'G000006785',\n       'G000006845', 'G000006865', 'G000006925', 'G000006965', 'G000007105',\n       ...\n       'G900155395', 'G900155405', 'G900155555', 'G900155635', 'G900155965',\n       'G900156305', 'G900156675', 'G900156765', 'G900156885', 'G900163845'],\n      dtype='object', name='OTUID', length=3771)] are in the [index]"_

It seems to be caused when attempting to merge two data frames, and the set of keys in one do not intersect at all with the keys in the other; it's like they're coming from two different datasets. Antonio and I think Justin might better at resolving this last issue than myself.

@antgonza @justinshaffer If you want to simply add to this branch or merge it to biocore:dev and add your changes I can help you.

### CODE REVIEW

**Comment by antgonza** [comment]
The test are not running cause you issue the PR to dev; could you change it to master?

**Comment by charles-cowart** [comment]
> The test are not running cause you issue the PR to dev; could you change it to master?

You got it! It should merge into the current main without issue. The functions and files are all new, and the sample metadata files being used aren't an existing type with recent changes e.g.: sample-sheet, pre-prep-file.

File: metapool/rescale_counts.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    converted_reads = pd.DataFrame(data=cols).T
+    return converted_reads, failed_colnames
+
+
+def convert_read_count_to_cell_count_per_g_input(df,
+                                                 metadata_features,
+                                                 prep_info_samples):
+    lengths = metadata_features.loc[df.index, "total_length"]
+    sample_weights = prep_info_samples.loc[df.index,
```
Comment (antgonza) [inline metapool/rescale_counts.py:42]: As far as I can tell, this is the problematic line (why the tests are failing), and looking at the code I think what's happening is that the values in `df` are transversed: basically, it's looking by features but it should be looking by sample. @charles-cowart, can you take a look?

**Comment by charles-cowart** [comment]
Update: Currently there is an issue with merging two different dataframes to generate the final output dataframe. The columns in metadata_samples_plasmid_sequences.txt do not intersect with those of the other two files used in the tests. It appears that the code and/or data to bridge this gap was not written. @antgonza  is working on it based on feedback from Justin.

**Comment by charles-cowart** [comment]
Quick update: Sent Amanda B. an invite to the project. Amanda will be working on this issue.

**Pull request closed without merging**